In [1]:
import tensorflow as tf
mnist=tf.keras.datasets.fashion_mnist

(training_images, training_labels),(test_images, test_labels)=mnist.load_data() # Step 1, DATA LOAD
training_images=training_images/255.0 #Normalization to increase computaional power Step 2
test_images=test_images/255.0
#Step 3 Training
#a. Architecture
#b. Compilation
#c. Train
model=tf.keras.models.Sequential([tf.keras.layers.Flatten(),#Flatten converts 3D,2D into 1D
                                  tf.keras.layers.Dense(512,activation=tf.nn.relu),#output of above layer becomes input of this second layer, so no need to give input shape here. .... Dense means densely connected layer .. 512 means it generate 512 outputs 
                                  tf.keras.layers.Dense(10,activation=tf.nn.softmax)])# This layer gets 512 inputs and gives 10 outputs. we use softmax function for multiclassification problems.its very important.
#Activation function in middle layers can be different as per your choice BUT relu function is commonly used the reason is "relu gives values greater than zero values but lower than zero values converted into only zero."
#tf.nn.     nn means neural network
# in last layer softmax returns probability of all numbers, higher number probablity gets its class.


4423680/4422102 [==============================] - 39s 9us/step


In [2]:
model.compile(optimizer =tf.keras.optimizers.Adam(),
             loss='sparse_categorical_crossentropy',
             metrics=['accuracy'])
model.fit(training_images, training_labels, epochs=5, batch_size=128)
model.evaluate(test_images, test_labels)#this is important step on unseen data ,it gives how good your model is.#Performance accuracy


Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 5s 87us/sample - loss: 0.5047 - accuracy: 0.8239
Epoch 2/5
60000/60000 [==============================] - 4s 74us/sample - loss: 0.3762 - accuracy: 0.8661
Epoch 3/5
60000/60000 [==============================] - 4s 72us/sample - loss: 0.3313 - accuracy: 0.8791
Epoch 4/5
60000/60000 [==============================] - 4s 66us/sample - loss: 0.3071 - accuracy: 0.8874
Epoch 5/5
60000/60000 [==============================] - 4s 65us/sample - loss: 0.2878 - accuracy: 0.8954
10000/1 [======================================================================================================================================================================================================================================================================================================================================================================================================================================================

[0.34674543092250826, 0.8759]

89us/sample - loss: 0.2735 - accuracy: 0.8759   # Model successfully trained with 87 % accuracy